libsupport/test/random.cpp

```diff
@@ -18,6 +19,34 @@ main() {
   for (std::thread& t : threads) {
     t.join();
   }
+  threads.clear();
+
+  // test to make sure seeded generators are deterministic
+  std::vector<katana::RandGenerator> generators;
+  generators.reserve(128);
+  for (int i = 0; i < 128; ++i) {
+    generators.emplace_back(katana::GetGenerator(8675309));
+  }
+
+  std::vector<std::string> results(128);
+  for (int i = 0; i < 128; ++i) {
+    threads.emplace_back([&generators, &results, i]() {
+      results[i] = katana::RandomAlphanumericString(12, &generators[i]);
+    });
+  }
+
+  for (std::thread& t : threads) {
+    t.join();
+  }
+
+  std::string first_val = results.front();
+  for (const auto& val : results) {
+    KATANA_LOG_VASSERT(
+        val == first_val,
+        "seeded rngs should output the same values, "
+        "got \"{}\" and \"{}\"",
```

Comment:
Oh, so all threads got the same string because they have thread-local generators that are identical?

Reply:
Almost, in this case I created my own thread local generators by seeding 128 with the same value.